src/components/sbb-journey-summary/sbb-journey-summary.tsx

```diff
@@ -0,0 +1,142 @@
+import { Component, h, JSX, Prop } from '@stencil/core';
+import { InterfaceJourneySummaryAttributes } from './sbb-journey-summary.custom';
+import isTomorrow from 'date-fns/isTomorrow';
+import { isToday, parseISO, format, intervalToDuration, isBefore, isValid } from 'date-fns';
+import {
+  i18nArrival,
+  i18nDeparture,
+  i18nDurationHour,
+  i18nDurationMinute,
+  i18nToday,
+  i18nTomorrow,
+  i18nWalkingDistanceArrival,
+  i18nWalkingDistanceDeparture,
+} from '../../global/i18n';
+import getDocumentLang from '../../global/helpers/get-document-lang';
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-journey-summary.scss',
+  tag: 'sbb-journey-summary',
+})
+export class SbbJourneySummary {
+  @Prop() public summaryConfig!: InterfaceJourneySummaryAttributes['config'];
+
+  private _currentLanguage = getDocumentLang();
+
+  private _formatTime(duration): JSX.Element {
+    return (
+      <span>
+        <time>{duration.hours}</time>
+        <span>{i18nDurationHour.single.short[this._currentLanguage]}</span>
+        <time>{duration.minutes}</time>
+        <span>{i18nDurationMinute.single.short[this._currentLanguage]}</span>
+      </span>
+    );
+  }
+
+  private _renderJourneyStart(departureTime: Date, arrivalTime: Date): JSX.Element {
+    let duration: Duration;
```

Comment:
Oliver would say: Try to avoid let :D 

Maybe we would extract the first if / else section in an other function